myriad-scheduler/src/test/java/com/ebay/myriad/scheduler/fgs/YarnNodeCapacityManagerSpec.groovy

```diff
@@ -0,0 +1,118 @@
+package com.ebay.myriad.scheduler.fgs
+
+import com.ebay.myriad.scheduler.TaskFactory
+import com.ebay.myriad.scheduler.TaskUtils
+import com.ebay.myriad.scheduler.yarn.interceptor.InterceptorRegistry
+import com.ebay.myriad.state.NodeTask
+import com.ebay.myriad.state.SchedulerState
+import org.apache.hadoop.yarn.api.records.ContainerState
+import org.apache.hadoop.yarn.server.resourcemanager.scheduler.event.NodeResourceUpdateSchedulerEvent
+import org.apache.hadoop.yarn.util.resource.Resources
+import org.apache.mesos.Protos
+
+/**
+ *
+ * Tests for YarnNodeCapacityManager
+ *
+ */
+class YarnNodeCapacityManagerSpec extends FGSTestBaseSpec {
+
+  def "No Containers Allocated Due To Mesos Offers"() {
+    given:
+    def yarnNodeCapacityMgr = getYarnNodeCapacityManager()
+
+    def host = "test_host"
+    def slaveId = Protos.SlaveID.newBuilder().setValue(host + "_slave_id").build()
+    def zeroNM = getRMNode(0, 0, host, slaveId)
+
+    // have a mesos offer before HB
+    def offer = addOfferToFeed(slaveId, host, 4, 4096)
+    offerLifecycleManager.markAsConsumed(offer)
+
+    //  2 containers before HB.
+    def fgsContainer1 = getFGSContainer(zeroNM, 1, 1, 1024, ContainerState.RUNNING)
+    def fgsContainer2 = getFGSContainer(zeroNM, 2, 1, 1024, ContainerState.RUNNING)
+    nodeStore.getNode(host).snapshotRunningContainers()
+
+    // Node's capacity set to match the size of 2 containers + mesos offers
+    yarnNodeCapacityMgr.setNodeCapacity(zeroNM, Resources.createResource(6144, 6))
+
+    // no new container allocations
+
+    when:
+    yarnNodeCapacityMgr.handleContainerAllocation(zeroNM)
+
+    then:
+    nodeStore.getNode(host).getNode().getRunningContainers().size() == 2  // 2 containers still running
```

Comment:
it is common to be more terse in groovy with chaining of properties... like this:

`nodeStore.getNode(host).node.runningContainers.size == 2  // 2 containers still running`